iroh-relay/src/dns/node_info.rs

```diff
@@ -200,12 +200,32 @@ impl NodeInfo {
     }
 }
 
+/// Parses a [`NodeId`] from a DNS domain name.
+///
+/// Splits the domain name into labels on each dot. Expects the first label to be
+/// [`IROH_TXT_NAME`] and the second label to be a z32 encoded [`NodeId`]. Ignores
+/// subsequent labels.
+///
+/// Returns a [`NodeId`] if parsed successfully, otherwise `None`.
+pub fn node_id_from_domain_name(name: &str) -> Option<NodeId> {
```

Comment:
This seems like a pkarr helper method.  Is there any reason to make all this pub?  Maybe this stuff was already pub, but I guess we haven't tried to make the pub API here much smaller yet.  If we don't want to reduce the amount of pub in this PR to keep concerns separate (I often prefer this in my own PRs) I'm also fine to leave this here for now.

Reply:
It was only used in `iroh::test_utils` - moved it there.